### PR TITLE
feat(pci.public.gateways): add onboarding for public gateways

### PR DESCRIPTION
**Pull request by vks-ovh**
Addition of onboading page for public gateways
  ref: MANAGER-9149

Signed-off-by: vikash singh <vikash.singh@corp.ovh.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | develop
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-9149
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Created a onboarding page for public gateways, this onboarding page will load conditionally.

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.routing.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('pci.projects.project.public-gateways.onboarding', {
+    url: '/onboarding',
+    component: 'pciProjectPublicGatewaysOnboarding',
+    redirectTo: (transition) =>
+      transition
+        .injector()
+        .getAsync('privateNetworks')
+        .then((privateNetworks) =>
+          privateNetworks.length < 0
```
Comment (vks-ovh) [inline packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.routing.js:10]: Here, I need to check `privateNetworks.length > 0`. I will take care of this in next commit where I have to change illustration image and some more condition for `create a public gateway` button. Waiting for some of the queries to get answered.

File: packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.controller.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    );
+  }
+
+  onAddPublicGatewayClick() {
+    this.trackPublicGateways('::onboarding::add');
+  }
+
+  onDocumentationClick(guide) {
+    this.trackPublicGateways(`onboarding::documentation::${guide.id}`);
+  }
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.controller.js:37]: I think there is one that is not good (::) at first.
Either they are too many for the first one, or they are missing for the second one.

Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.controller.js:37]: I think there is one that is not good (::) at first.
Either they are too many for the first one, or they are missing for the second one.

Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.controller.js:37]: I think there is one that is not good (::) at first.
Either they are too many for the first one, or they are missing for the second one.

**Review by JeremyDec — CHANGES_REQUESTED** [review]
Hello @vks-ovh can you check please, there are error on test/build from your CDS run

File: packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.constants.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+export const GUIDES = [
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.constants.js:1]: No others Subs links ?

File: packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.controller.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            `pci_projects_project_public_gateways_onboarding_guides_${guide.id}_description`,
+          ),
+        },
+      ],
+      [],
+    );
+  }
+
+  onAddPublicGatewayClick() {
+    this.trackPublicGateways('::onboarding::add');
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.controller.js:32]: I have try to locate `trackPublicGateways` function but i don't success.
May be implementation forget ?

File: packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.service.js [hunk 5]
```diff
@@ -0,0 +1 @@
+export default class publigGatewaysOnboardingServiceClass {}
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.service.js:1]: Don't implement something which is not required.

File: packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/translations/Messages_fr_FR.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
+{
+  "pci_projects_project_public_gateways_onboarding": "Public Gateway",
+  "pci_projects_project_public_gateways_onboarding_content1": "Une instance entièrement dédiée installée dans un réseau privé peut nécessiter l’accès à des ressources internet. Public Gateway est un moyen de connexion sécurisé, sans exposition directe de l’instance à Internet.",
+  "pci_projects_project_public_gateways_onboarding_content2": "Une application web exposée à Internet doit posséder une adresse IP. Pour cela, une IP flottante constitue la meilleure solution. Il s'agit d'une IP flexible et agnostique, pouvant être déplacée entre des services Public Cloud. Pour une telle utilisation, elle doit être configurée sur une Public Gateway et directement liée à une instance privée ou au Load Balancer. Et ce, pour une meilleure évolutivité du service.",
+  "pci_projects_project_public_gateways_onboarding_action_label": "Créer une Public Gateway",
+  "pci_projects_project_public_gateways_onboarding_guides_creat-private-network-with-gateway_title": "Créer un réseau privé avec Public gateway depuis l'espace client OVHcloud",
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/translations/Messages_fr_FR.json:6]:
```suggestion
  "pci_projects_project_public_gateways_onboarding_guides_creat-private-network-with-gateway_title": "Créer un réseau privé avec Public Gateway depuis l'espace client OVHcloud",
```

File: packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/translations/Messages_fr_FR.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+  "pci_projects_project_public_gateways_onboarding_guides_creat-private-network-with-gateway_description": "",
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/translations/Messages_fr_FR.json:7]: Empty ? We need it ?

File: packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/translations/Messages_fr_FR.json [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+  "pci_projects_project_public_gateways_onboarding_guides_attach_pci_floating_ip_to_instance_title": "Associer une IP flottante Public Cloud à une instance",
+  "pci_projects_project_public_gateways_onboarding_guides_attach_pci_floating_ip_to_instance_description": "",
+  "pci_projects_project_public_gateways_onboarding_guides_start_load_balancer_service_title": "Créer un Load Balancer HTTP/HTTPS depuis l'espace client OVHcloud",
+  "pci_projects_project_public_gateways_onboarding_guides_start_load_balancer_service_description": ""
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/translations/Messages_fr_FR.json:11]: Same as here:
https://github.com/ovh/manager/pull/7020/files#r866587325

File: packages/manager/modules/pci/src/projects/project/sidebar/sidebar.constant.js [hunk 9]
```diff
@@ -112,7 +112,7 @@ export const getMenu = ({ DBAAS_LOGS_URL }) => [
       {
         id: 'public-gateways',
         options: {
-          state: 'pci.projects.project.publicGateways',
+          state: 'pci.projects.project.public-gateways',
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/sidebar/sidebar.constant.js:115]: 👍

File: packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.constants.js (hunk 3 above)
Comment (vks-ovh) [reply packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.constants.js:1]: For now we have english only.

File: packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.controller.js (hunk 2 above)
Comment (vks-ovh) [reply packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.controller.js]: fixed, it was extra.

File: packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.service.js (hunk 5 above)
Comment (vks-ovh) [reply packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.service.js]: Removed as it was not required.

File: packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/translations/Messages_fr_FR.json (hunk 7 above)
Comment (vks-ovh) [reply packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/translations/Messages_fr_FR.json]: Removed as we don't have any description as of now, if it comes will add it again.

Comment (mohammed-zahaf) [reply packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/translations/Messages_fr_FR.json]: Not issue if you know you will complete it before we prod.

File: packages/manager/modules/pci/src/projects/project/public-gateways/public-gateways.controller.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
+const CONTAINER_NAME = 'pci.projects.project.public-gateways';
+
+export default class {
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/public-gateways/public-gateways.controller.js:3]: **nitpick:** Better to use named class.
```suggestion
export default class PublicGatewaysController {
```

File: packages/manager/modules/pci/src/projects/project/public-gateways/public-gateways.routing.js [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+          publicGateways.length === 0
+            ? { state: 'pci.projects.project.public-gateways.onboarding' }
+            : false,
+        ),
+    resolve: {
+      breadcrumb: /* @ngInject */ ($translate) =>
+        $translate.instant('pci_projects_project_public_gateway_title'),
+      publicGateways: /* @ngInject */ (PciPublicGatewaysService, projectId) => {
+        return PciPublicGatewaysService.getPublicGateways(projectId);
+      },
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/public-gateways/public-gateways.routing.js:19]:
```suggestion
      publicGateways: /* @ngInject */ (PciPublicGatewaysService, projectId) => PciPublicGatewaysService.getPublicGateways(projectId),
```

**Review by anooparveti — APPROVED** [review]

File: packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.constants.js [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+  {
+    id: 'creat-private-network-with-gateway',
+    link:
```
Comment (radireddy) [inline packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.constants.js:4]: You need to add support for all subs even though you have only one sub as of now. Please refer to Kubernetes onboarding page for code example.

Comment (vks-ovh) [reply packages/manager/modules/pci/src/projects/project/public-gateways/onboarding/onboarding.constants.js:4]: @radireddy I have taken care of this in config page as suggested by you.

**Review by ghyenne — APPROVED** [review]

**Review by radireddy — APPROVED** [review]

**Pull request merged**